core/iwasm/common/arch/invokeNative_aarch64.s

```diff
@@ -2,6 +2,7 @@
  * Copyright (C) 2019 Intel Corporation.  All rights reserved.
  * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
  */
+#if (defined(BH_PLATFORM_DARWIN) && defined(__aarch64__) || !defined(BH_PLATFORM_DARWIN))
```

Comment:
When building a universal binary, the compiler compiles each source file twice, once for each architecture. This can't be handled by conditionally including assembly files as we currently do in `iwasm_common.cmake`. If we only include the file for one architecture (without any macros), the linker will complain about missing symbols for the architecture which we're not including an assembly file for and the compiler will complain about invalid assembly instructions for the assembly file we do include (since it will be compiled for _both_ architectures).

If we can't use macros here, an alternative might be to add two assembly files `invokeNative_osx_universal.s` and `invokeNative_osx_universal_simd.s`.

with content

```
#if defined(__aarch64__)
// asm from invokeNative_aarch64.s here
#else
// asm from invokeNative_em64.s here
#endif
```
and similar for the simd asm file. This is what apple suggest: [Wrap Platform Specific Code with Conditional Compilation Macros](https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary#Wrap-Platform-Specific-Code-with-Conditional-Compilation-Macros).

Reply:
Can we only add a file invokeNative_osx_universal.s? like:
```C
#if defined(BUILD_TARGET_AARCH64)
#if WASM_ENABLE_SIMD == 0
#include "invokeNative_aarch64.s"
#else
#include "invokeNative_aarch64_simd.s"
#endif
#else
#if WASM_ENABLE_SIMD == 0
#include "invokeNative_em64.s"
#else
#include "invokeNative_em64_simd.s"
#endif
#endif
```